index.js

```diff
@@ -0,0 +1,241 @@
+        });
+
+    });
+});
+const phoenixClient = new Client({
+    intents: [
+        GatewayIntentBits.Guilds,
+        GatewayIntentBits.GuildMessages,
+        GatewayIntentBits.MessageContent,
+        GatewayIntentBits.GuildMembers,
+    ],
+});
+
+async function handleRoleRemoval() {
+    const now = new UTCDate();
+    const events = await phoenix.getEvents();
+    const nextEventTimeOffsetted = events
+        .map(event => subMonths(new UTCDate(event["start_time"] * 1000), timeBeforeNextEventRemove))
+        .find(time => time.getTime() > now.getTime());
+    if (!nextEventTimeOffsetted || differenceInHours(nextEventTimeOffsetted, now) > 25) {
+        console.info("Skipping scheduled role removal");
+        return;
+    }
+
+    console.log("Removing roles");
+    removeAllRoles();
+}
+setInterval(() => {
+    handleRoleRemoval();
+}, 8.64e+7); // 1 day
+
+
+async function removeAllRoles() {
+    const guild = phoenixClient.guilds.cache.get(phoenixGuildId);
+    try {
+        let crews = await Promise.all((await phoenix.Crew.getCrews()));
+        crews.forEach(async (Crew) => {
+            let allCrews = await phoenix.Crew.getCrew(Crew.uuid);
+            let readableCrew = allCrews.positions;
+            readableCrew.forEach(async (position) => {
+                if (position.chief === true) {
+                    position.position_mappings.forEach(async (mapping) => {
+                        let discordUser = await phoenix.User.getDiscordMapping(mapping.user.uuid);
+                        if (discordUser != null && discordUser.discord_id != null) {
+                            let roleVar = guild.roles.cache.find(role => role.name === "Gruppeleder");
+                            let member = await guild.members.fetch(discordUser.discord_id);
+                            await member.roles.remove(roleVar);
+                        }
+                    })
+                } else {
+                    position.position_mappings.forEach(async (mapping) => {
+                        let discordUser = await phoenix.User.getDiscordMapping(mapping.user.uuid);
+                        if (discordUser != null && discordUser.discord_id != null) {
+                            let roleVar = guild.roles.cache.find(role => role.name === "Crew");
+                            let member = await guild.members.fetch(discordUser.discord_id);
+                            await member.roles.remove(roleVar);
+                        }
+                    })
+                }
+            });
+        });
+    } catch (error) {
+        console.error("an error occured while deleting roles:", error);
+    }
+
+}
+
+async function updateRoles() {
+    //change this
+    const guild = phoenixClient.guilds.cache.get(phoenixGuildId);
+    let crews = await Promise.all((await phoenix.Crew.getCrews()));
+    crews.forEach(async (Crew) => {
+        let allCrews = await phoenix.Crew.getCrew(Crew.uuid);
+        let readableCrew = allCrews.positions;
```

Comment:
There's no need for these variables to be mutable.
```suggestion
        const allCrews = await phoenix.Crew.getCrew(Crew.uuid);
        const readableCrew = allCrews.positions;
```